test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1

```diff
@@ -85,6 +90,11 @@ Describe "Clear-Content cmdlet tests" -Tags "CI" {
       Get-Content -Path "TestDrive:/$file3" | Should -BeExactly $content2
       Get-Content -Path "TestDrive:/$file3" -Stream $streamName | Should -BeNullOrEmpty
     }
+    It "Should not error when targeting an existing stream on a directory." -Skip:(!$IsWindows) {
+      { Set-Content -Path "TestDrive:/$dirName" -Stream $streamName -Value $streamContent } | Should -Not -Throw
+      Get-Content -Path "TestDrive:/$dirName" -Stream $streamName | Should -BeExactly $streamContent
+      { Clear-Content -Path TestDrive:\$dirName -Stream $streamName -ErrorAction Stop } | Should -Not -Throw
```

Comment:
We prefer positive tests over "Should -Not -Throw".

Also Set-Content tests should be in Set-Content test file.

Reply:
I'm trying to ensure that the stream actually exists before continuing the test. Of the file systems that Windows provides, only NTFS actually supports alternate data streams. This means that it's impossible to test data streams on ReFS et al.  But I can't change all the tests for alternate data streams in this PR, because they're out of scope.

(This means that literally every unit test that Powershell has that relates to streams is poorly-written because it relies on external environmental factors which cannot be guaranteed. The Set-Content | Should -Not -Throw is me starting to figure out what is supposed to happen here.)

I'd like to keep the `Clear-Content -ErrorAction Stop | -Should -Not -Throw` (as a positive test that it does not error in the environment that was set up).  Can I do that and follow it up with `Get-Content -Path "TestDrive:/$dirName -Stream $streamName"` that tests that it still exists, but its size is zero?  Because that's what needs to be tested for here.

Reply:
I suggest:
```powershell
    It "Should not error when targeting an existing stream on a directory." -Skip:(!$IsWindows) {
      Set-Content -Path "TestDrive:\$dirName" -Stream $streamName -Value $streamContent
      Get-Content -Path "TestDrive:\$dirName" -Stream $streamName | Should -BeExactly $streamContent
      Clear-Content -Path TestDrive:\$dirName -Stream $streamName
      Get-Content -Path "TestDrive:\$dirName" -Stream $streamName | Should -BeNullOrEmpty
```